### PR TITLE
ActionMenu strict typings

### PR DESCRIPTION
**Pull request by danethurber**
and various final touches

### CODE REVIEW

File: packages/actionmenu/src/react/index.tsx [hunk 1]
```diff
@@ -1,62 +1,80 @@
...
+      children,
+      ...rest
+    },
+    forwardedRef
+  ) => {
+    const ref = React.useRef<HTMLUListElement | null>(null)
+    useImperativeHandle<HTMLUListElement | null, HTMLUListElement | null>(
+      forwardedRef,
+      () => ref.current
+    )
```
Comment (danethurber) [inline packages/actionmenu/src/react/index.tsx:76]: found a way to avoid casting the ref...not sure it's any better 🤷

Comment (EdwardIrby) [reply packages/actionmenu/src/react/index.tsx:76]: looks a bit better

**Review by EdwardIrby — APPROVED** [review]

File: packages/actionmenu/src/react/item.tsx [hunk 2]
```diff
@@ -47,15 +52,29 @@ const styles: { [name: string]: StyleFn } = {
...
+  nested?: ReactNode
+  origin?: ValueOf<typeof origins>
+  value?: ReactText
+  onClick?: (event: MouseEvent, value?: ReactText) => void
+}
+
+interface AnchorProps
+  extends BaseItemProps,
+    Omit<HTMLPropsFor<'a'>, 'onClick'> {
+  tagName?: 'a'
```
Comment (jaketrent) [inline packages/actionmenu/src/react/item.tsx:66]: Why introduce another patten for this and not use the existence of href or the renderProp patterns that we have so far?

File: packages/actionmenu/tsconfig.json [hunk 3]
```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "allowJs": true,
     "rootDir": null,
-    "jsx": "react"
+    "jsx": "react",
+    "strict": true
```
Comment (jaketrent) [inline packages/actionmenu/tsconfig.json:7]: nice catch!

File: packages/actionmenu/src/react/item.tsx (hunk 2 above)
Comment (danethurber) [reply packages/actionmenu/src/react/item.tsx:66]: to avoid a breaking change. this is the pattern that was used in ActionMenu - i just added the typings

**Pull request merged**